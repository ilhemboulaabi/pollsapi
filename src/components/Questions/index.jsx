import React from 'react';
import {
  Grid,
  LinearProgress,
  Box,
} from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';
import PropTypes from 'prop-types';
import Question from '../Question';

const Questions = ({ questions, loading }) => (
  loading || (questions === undefined || questions === null || questions.length === 0)
    ? (
      <Box>
        <LinearProgress />
        <Alert severity="info">No Data available!</Alert>
      </Box>
    )
    : (
      <Grid container spacing={2}>
        {
        questions && questions.map((question) => (
          <Question
            key={question.url}
            question={question}
          />
        ))
        }
      </Grid>
    )
);

export default Questions;

Questions.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  questions: PropTypes.array,
  loading: PropTypes.bool,
};
Questions.defaultProps = {
  questions: [],
  loading: false,
};
