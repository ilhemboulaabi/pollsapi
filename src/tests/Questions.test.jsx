import * as React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Questions from '../components/Questions';

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    questions: [],
    isQuestionsLoading: false,
    actions: {
      questions: {
        loadQuestions: jest.fn(),
        searchQuestion: jest.fn(),
      },
    },
  };

  const wrapper = shallow(
    <Questions {...props} />,
    {
      disableLifecycleMethods: true,
    },
  );

  return {
    props,
    wrapper,
  };
};

describe('Questions', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
