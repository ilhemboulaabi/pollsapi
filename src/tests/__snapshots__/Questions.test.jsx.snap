// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions should render self 1`] = `
<Styled(MuiBox)>
  <WithStyles(ForwardRef(LinearProgress)) />
  <WithStyles(ForwardRef(Alert))
    severity="info"
  >
    No Data available!
  </WithStyles(ForwardRef(Alert))>
</Styled(MuiBox)>
`;
